### ISSUE TITLE
Renable Recipe Jank" causes worlds to not load with Quiltloader

### ISSUE THREAD

**Issue by Shibva** [TheWinABagel/Zenith#33]
whenever you load a world you get tossed an error that kicks you from the world and returns you to the main menu (or server screen for errors of this kind)

I dont have the means to open this on this computer but this should be the log that contains the error
[2023-05-30-2.log.gz](https://github.com/TheWinABagel/Zenith/files/11608514/2023-05-30-2.log.gz)

also just curios, why did you renable it? (also I saw the changelog beforehand; you are forgiven X)  )

**Comment by TheWinABagel** [comment]
It wasn't rendering the recipes properly on servers, the potions in specific (nbt wasnt being loaded) 
The recipes still worked though, its just people couldnt figure out how to craft some things (#29)
I tried to figure out a fix that would continue using NBT crafting but I couldn't figure it out, and there was a pack update waiting on me only so I felt a little rushed, I hope to be able to get a version that both works on servers and quilt in the future

**Comment by Shibva** [comment]
I don't think there's a specific way you can do datapacks with NBT data… You'll have to find a fancy way around it. Hope you can find a way to make it work with quilt :-)

**Comment by Shibva** [comment]
I don't think there's a specific way you can do datapacks with NBT data…

I think that requires Mixins or recipe registry to allow for NBT specific operations

Take my birth of a bit of salt, I really don't know much about this sort of thing, but I do know that it seems like that datapaks lack some sort of NBT specific operation

**Comment by Shibva** [comment]
I think theres a way you can still use datapacks to work wtih NBT; however you might need to create a custom crafting method that works in the crafting table that allows for NBT data and stuff

or, there may be a way to have JEI/REI read potions correctly using their API; I will tell you that the invalid potions and charms also render on survival worlds as well

**Comment by Shibva** [comment]
If it helps with the current issue; heres a newer crash report quilt gave when running the 0.6.4 to have an understanding what may have gone wrong this time around

https://gist.github.com/Shibva/b27491513569b7dd9b1968388cf1c363

**Comment by Shibva** [comment]
I forgot to mention; this issue has resulted from the world load to a game crash

**Comment by Shibva** [comment]
So I decided to try this with fabric and got a crash on startup. Ill make a separate issue wtih this bug;

on another note; when I used 0.6.2; the jank dint fix anything wtih the potion badges; however it did fix the bookshelf issue specifically
<img width="650" alt="image" src="https://github.com/TheWinABagel/Zenith/assets/38770780/9683c758-aee8-451b-a676-3bb4a6159b2a"> I recommend maybe reversing the jank stuff and allowing for only specific recipes that may need special operations be used


As for how to make this possibly work for Quilt and Fabric: I found that Extra Alchemy has something similar called potion rings and it seems like what their doing for crafting works both in Farbic and Quilt; the code for how it handles the crafting can be found [here](https://github.com/zabi94/ExtraAlchemy/blob/fabric-main/src/main/java/zabi/minecraft/extraalchemy/crafting/PotionRingRecipe.java#L14)


Hope this helps @TheWinABagel

**Comment by Shibva** [comment]
### On ANOTHER NOTE:
I have good news; the fabric issue wtih it kicking me on join is not happing anymore on the intinal reported 0.6.2 version anymore! I think it was because I was using a version of quilt that was running version 0.18.1 of Fabricloader

ill continue this issue but as a new one as well to avoid confusion

**Comment by Shibva** [comment]
I stand corrected

though I think I will make a new issue to discuss this PROPERLY
